### PR TITLE
Reduce scope of `com.sap.cloud.security:java-api` to `provided`

### PR DESCRIPTION
**Pull request by newtork**
## Context
 

The dependency doesn't necessarily need to be at runtime for all cases.
It's only required IF the `SecurityContext` class is available at all.

## Definition of Done

- [x] Functionality scope stated & covered
- [ ] E2E tested
- [x] ~Error handling created / updated & covered by the tests above~
- [x] ~Documentation updated~
- [x] ~Release notes updated~

### CODE REVIEW

File: cloudplatform/security/pom.xml [hunk 1]
```diff
@@ -70,6 +66,11 @@
 			<artifactId>httpcore</artifactId>
 		</dependency>
 		<!-- scope "provided" -->
+		<dependency>
+			<groupId>com.sap.cloud.security</groupId>
+			<artifactId>java-api</artifactId>
+			<scope>provided</scope>
```
Comment (MatKuhr) [inline cloudplatform/security/pom.xml:72]: (minor) alternatively we could use `optional` to indicate this module also works without this dependency

File: cloudplatform/security/src/main/java/com/sap/cloud/sdk/cloudplatform/security/SecurityContextThreadContextDecorator.java [hunk 2]
```diff
@@ -31,9 +30,18 @@ public class SecurityContextThreadContextDecorator implements ThreadContextDecor
     @Override
     public <T> Callable<T> decorateCallable( @Nonnull final Callable<T> callable )
     {
+        try {
+            getClass().getClassLoader().loadClass("com.sap.cloud.security.token.SecurityContext");
+        }
+        catch( final Exception e ) {
```
Comment (MatKuhr) [inline cloudplatform/security/src/main/java/com/sap/cloud/sdk/cloudplatform/security/SecurityContextThreadContextDecorator.java:36]:
```suggestion
        catch( final ClassNotFoundException e ) {
```

Also, apparently one can [skip instantiation of the class](https://www.baeldung.com/java-check-class-exists#telling-classforname-to-skip-initialization).

**Review by MatKuhr — APPROVED** [review]
pending e2e testing

**Comment by newtork** [comment]
Blocked until E2E test runs again.

**Review by MatKuhr — APPROVED** [review]
Manually E2E tested with the JCo E2E test, works as expected 👍🏻

**Pull request merged**